### ISSUE TITLE
Should we change the category filters for AppInsights and disable Sampling by default?

### ISSUE THREAD

**Issue by fume** [microsoft/SPID-and-Digital-Identity-Enabler#18]
**TL;DR;**  
I'm willing to reduce the logs sent to AppInsights changing the category filters. At the same time i would like to disable the AdaptiveSampling by default to avoid loosing important logs which are mandatory for AgID.

**Details**  
With the actual default configuration, we send Information logs to AppInsights for every single category:
```json
{
	"Logging": {
		"ApplicationInsights": {
			"LogLevel": {
				"Default": "Information"
			}
		}
       }
}
```

With this configuration we are also sending framework logs such as the following, which are not really useful and could produce a lot of data-ingestion
![image](https://user-images.githubusercontent.com/2969850/172844645-690cef86-74ea-4651-82e0-002840d6aabb.png)

Should we change the category filters for AppInsights to send Information logs and above for the `Microsoft.SPID` cateogory and sending only Warning and above logs for the `Default` category? Something like this:
```json
{
	"Logging": {
		"ApplicationInsights": {
			"LogLevel": {
				"Default": "Warning",
                                "Microsoft.SPID": "Information"
			}
		}
       }
}
```
We could increase the verbosity at any-time by overriding the setting via Configuration.
I would like to reduce the telemetry sent to AppInsights because AgID requires to store logs for 24 months. We can change the retention of the Application Insights instance to achieve the objective (via the linked LAW or on the AppInsights instance directly), but it has an extra-cost per-GB/per-Month.

Moreover, we have Adaptive Sampling enabled by default. Whene there are a lot of concurrent requets towards the SPIDProxy, sampling is applied hence we loose log details but still have correct numbers (thanks to the itemCount). I would like to disable Sampling by default since we could sample a log which is required by AgID and which we should retain for 24 months.
We can do so by adding the following in the appsettings.json, as also shown here https://docs.microsoft.com/en-us/azure/azure-monitor/app/asp-net-core#configuration-recommendation-for-microsoftapplicationinsightsaspnetcore-sdk-2150-and-later:
```json
{
    "ApplicationInsights": {
        "EnableAdaptiveSampling": false
    }
}
```

What do you think @tommasodotNET, @MarcoZama, @PaoloCastAway ?

**Comment by MarcoZama** [comment]
@fume Looks good for me!
